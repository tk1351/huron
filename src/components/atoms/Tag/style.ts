import { css } from '@emotion/react'

export const tag = css({
  backgroundColor: '#066163',
  borderRadius: '10%',
  padding: '0.2rem',
  textDecoration: 'none'
})
