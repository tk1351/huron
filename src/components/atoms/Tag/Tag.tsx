import { FC } from 'react'
import { tag } from './style'

type Props = {
  title: string
  href: string
}

export const Tag: FC<Props> = ({ title, href }) => (
  <a css={tag} href={href}>
    {title}
  </a>
)
