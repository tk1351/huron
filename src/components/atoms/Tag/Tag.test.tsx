import * as renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'
import { Tag } from './Tag'

const expectedTitle = 'tag'
const expectedHref = '#'

const Component = <Tag title={expectedTitle} href={expectedHref} />

describe('Testing Tag Component', () => {
  describe('Render', () => {
    it('正しくTagコンポーネントが表示されていること', () => {
      const App = renderer.create(Component)
      expect(App).toMatchSnapshot()
    })
  })

  describe('Props', () => {
    describe('title', () => {
      it('タグのタイトルが正しく渡っていること', () => {
        render(Component)
        expect(screen.getByRole('link')).toHaveTextContent(expectedTitle)
      })
    })

    describe('href', () => {
      it('タグのhrefに指定した値が正しく渡っていること', () => {
        render(Component)
        expect(screen.getByRole('link')).toHaveAttribute('href', expectedHref)
      })
    })
  })
})
