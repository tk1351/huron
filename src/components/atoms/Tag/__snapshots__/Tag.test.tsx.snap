// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Tag Component Render 正しくTagコンポーネントが表示されていること 1`] = `
<a
  css={
    Object {
      "map": undefined,
      "name": "1dz688v",
      "next": undefined,
      "styles": "background-color:#066163;border-radius:10%;padding:0.2rem;text-decoration:none;",
      "toString": [Function],
    }
  }
  href="#"
>
  tag
</a>
`;
